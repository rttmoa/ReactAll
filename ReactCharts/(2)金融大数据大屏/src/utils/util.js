export function debounce(fun, delay) {
  let timer
  return function (...args) {
    if (timer) {
      window.clearTimeout(timer)
    }
    timer = setTimeout(() => fun.apply(this, args), delay)
  }
}

export function moment(timestaps = null) {
  const d = timestaps
    ? new Date(timestaps)
    : new Date()
  const [year, mounth, day, hour, minute, second] = [
    d.getFullYear(),
    (d.getMonth() + 1).toString().padStart(2, '0'),
    d.getDate().toString().padStart(2, '0'),
    d.getHours().toString().padStart(2, '0'),
    d.getMinutes().toString().padStart(2, '0'),
    d.getSeconds().toString().padStart(2, '0'),
  ]

  return {
    format(patterns = 'yyyy-MM-DD hh:mm:ss') {
      return patterns
        .replace(/yyyy/g, year)
        .replace(/MM/g, mounth)
        .replace(/DD/g, day)
        .replace(/hh/g, hour)
        .replace(/mm/g, minute)
        .replace(/ss/g, second)
    }
  }
}

export function formatMoney(money, decimal = 2) {
  money = Number(money)
  if (money === 0 && Number.isNaN()) return money
  return money.toFixed(decimal).replace(/\B(?=(\d{3})+(?!\d))/g, ',')
}

export function convertData(data = []) {
  const res = [];
  for (const item of data) {
    const geoCoord = geoCoordMap[item.name];
    if (geoCoord) {
      res.push({
        name: item.name,
        value: geoCoord.concat(item.value)
      });
    }
  }
  return res;
}

export const geoCoordMap = {
  上海: [119.1803, 31.2891],
  福建: [119.4543, 25.9222],
  重庆: [108.384366, 30.439702],
  北京: [116.4551, 40.2539],
  辽宁: [123.1238, 42.1216],
  河北: [114.4995, 38.1006],
  天津: [117.4219, 39.4189],
  山西: [112.3352, 37.9413],
  陕西: [109.1162, 34.2004],
  甘肃: [103.5901, 36.3043],
  宁夏: [106.3586, 38.1775],
  青海: [101.4038, 36.8207],
  新疆: [87.9236, 43.5883],
  西藏: [91.11, 29.97],
  四川: [103.9526, 30.7617],
  吉林: [125.8154, 44.2584],
  山东: [117.1582, 36.8701],
  河南: [113.4668, 34.6234],
  江苏: [118.8062, 31.9208],
  安徽: [117.29, 32.0581],
  湖北: [114.3896, 30.6628],
  浙江: [119.5313, 29.8773],
  内蒙古: [110.3467, 41.4899],
  江西: [116.0046, 28.6633],
  湖南: [113.0823, 28.2568],
  贵州: [106.6992, 26.7682],
  云南: [102.9199, 25.4663],
  广东: [113.12244, 23.009505],
  广西: [108.479, 23.1152],
  海南: [110.3893, 19.8516],
  黑龙江: [127.9688, 45.368],
  台湾: [121.4648, 25.563],
  南海诸岛: [114.252615, 15.86029]
};