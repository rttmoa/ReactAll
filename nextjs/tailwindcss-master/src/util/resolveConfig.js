import negateValue from './negateValue'
import corePluginList from '../corePluginList'
import configurePlugins from './configurePlugins'
import colors from '../public/colors'
import { defaults } from './defaults'
import { toPath } from './toPath'
import { normalizeConfig } from './normalizeConfig'
import isPlainObject from './isPlainObject'
import { cloneDeep } from './cloneDeep'
import { parseColorFormat } from './pluginUtils'
import { withAlphaValue } from './withAlphaVariable'
import toColorValue from './toColorValue'

function isFunction(input) {
  return typeof input === 'function'
}

function mergeWith(target, ...sources) {
  let customizer = sources.pop()

  for (let source of sources) {
    for (let k in source) {
      let merged = customizer(target[k], source[k])

      if (merged === undefined) {
        if (isPlainObject(target[k]) && isPlainObject(source[k])) {
          target[k] = mergeWith({}, target[k], source[k], customizer)
        } else {
          target[k] = source[k]
        }
      } else {
        target[k] = merged
      }
    }
  }

  return target
}

const configUtils = {
  colors,
  negative(scale) {
    // TODO: Log that this function isn't really needed anymore?
    return Object.keys(scale)
      .filter((key) => scale[key] !== '0')
      .reduce((negativeScale, key) => {
        let negativeValue = negateValue(scale[key])

        if (negativeValue !== undefined) {
          negativeScale[`-${key}`] = negativeValue
        }

        return negativeScale
      }, {})
  },
  breakpoints(screens) {
    return Object.keys(screens)
      .filter((key) => typeof screens[key] === 'string')
      .reduce(
        (breakpoints, key) => ({
          ...breakpoints,
          [`screen-${key}`]: screens[key],
        }),
        {}
      )
  },
}

function value(valueToResolve, ...args) {
  return isFunction(valueToResolve) ? valueToResolve(...args) : valueToResolve
}

function collectExtends(items) {
  return items.reduce((merged, { extend }) => {
    return mergeWith(merged, extend, (mergedValue, extendValue) => {
      if (mergedValue === undefined) {
        return [extendValue]
      }

      if (Array.isArray(mergedValue)) {
        return [extendValue, ...mergedValue]
      }

      return [extendValue, mergedValue]
    })
  }, {})
}

function mergeThemes(themes) {
  return {
    ...themes.reduce((merged, theme) => defaults(merged, theme), {}),

    // In order to resolve n config objects, we combine all of their `extend` properties
    // into arrays instead of objects so they aren't overridden.
    extend: collectExtends(themes),
  }
}

function mergeExtensionCustomizer(merged, value) {
  // When we have an array of objects, we do want to merge it
  if (Array.isArray(merged) && isPlainObject(merged[0])) {
    return merged.concat(value)
  }

  // When the incoming value is an array, and the existing config is an object, prepend the existing object
  if (Array.isArray(value) && isPlainObject(value[0]) && isPlainObject(merged)) {
    return [merged, ...value]
  }

  // Override arrays (for example for font-families, box-shadows, ...)
  if (Array.isArray(value)) {
    return value
  }

  // Execute default behaviour
  return undefined
}

function mergeExtensions({ extend, ...theme }) {
  return mergeWith(theme, extend, (themeValue, extensions) => {
    // The `extend` property is an array, so we need to check if it contains any functions
    if (!isFunction(themeValue) && !extensions.some(isFunction)) {
      return mergeWith({}, themeValue, ...extensions, mergeExtensionCustomizer)
    }

    return (resolveThemePath, utils) =>
      mergeWith(
        {},
        ...[themeValue, ...extensions].map((e) => value(e, resolveThemePath, utils)),
        mergeExtensionCustomizer
      )
  })
}

/**
 *
 * @param {string} key
 * @return {Iterable<string[] & {alpha: string | undefined}>}
 */
function* toPaths(key) {
  let path = toPath(key)

  if (path.length === 0) {
    return
  }

  yield path

  if (Array.isArray(key)) {
    return
  }

  let pattern = /^(.*?)\s*\/\s*([^/]+)$/
  let matches = key.match(pattern)

  if (matches !== null) {
    let [, prefix, alpha] = matches

    let newPath = toPath(prefix)
    newPath.alpha = alpha

    yield newPath
  }
}

function resolveFunctionKeys(object) {
  // theme('colors.red.500 / 0.5') -> ['colors', 'red', '500 / 0', '5]

  const resolvePath = (key, defaultValue) => {
    for (const path of toPaths(key)) {
      let index = 0
      let val = object

      while (val !== undefined && val !== null && index < path.length) {
        val = val[path[index++]]

        let shouldResolveAsFn =
          isFunction(val) && (path.alpha === undefined || index <= path.length - 1)

        val = shouldResolveAsFn ? val(resolvePath, configUtils) : val
      }

      if (val !== undefined) {
        if (path.alpha !== undefined) {
          let normalized = parseColorFormat(val)

          return withAlphaValue(normalized, path.alpha, toColorValue(normalized))
        }

        if (isPlainObject(val)) {
          return cloneDeep(val)
        }

        return val
      }
    }

    return defaultValue
  }

  Object.assign(resolvePath, {
    theme: resolvePath,
    ...configUtils,
  })

  return Object.keys(object).reduce((resolved, key) => {
    resolved[key] = isFunction(object[key]) ? object[key](resolvePath, configUtils) : object[key]

    return resolved
  }, {})
}

function resolvePlugins(configs) {
  let pluginGroups = []
  let allConfigs = []

  for (let config of configs) {
    allConfigs.push(config)

    let plugins = []

    for (let plugin of config?.plugins ?? []) {
      // TODO: If we want to support ESM plugins then a handful of things will have to become async
      if (typeof plugin === 'string') {
        // If the plugin is specified as a string then it's just the package name
        plugin = require(plugin)
        plugin = plugin.default ?? plugin
      } else if (Array.isArray(plugin)) {
        // If the plugin is specified as an array then it's a package name and optional options object
        // [name] or [name, options]
        let [pkg, options = undefined] = plugin
        plugin = require(pkg)
        plugin = plugin.default ?? plugin
        plugin = plugin(options)
      }

      if (plugin.__isOptionsFunction) {
        plugin = plugin()
      }

      // We're explicitly skipping registering child plugins
      // This will change in v4
      let [, childConfigs] = resolvePlugins([plugin?.config ?? {}])

      plugins.push(plugin)
      allConfigs.push(...childConfigs)
    }

    pluginGroups.push(plugins)
  }

  // Reverse the order of the plugin groups
  // This matches the old `reduceRight` behavior of the old `resolvePluginLists`
  // Why? No idea.
  let plugins = pluginGroups.reverse().flat()

  return [plugins, allConfigs]
}

function resolveCorePlugins(corePluginConfigs) {
  const result = [...corePluginConfigs].reduceRight((resolved, corePluginConfig) => {
    if (isFunction(corePluginConfig)) {
      return corePluginConfig({ corePlugins: resolved })
    }
    return configurePlugins(corePluginConfig, resolved)
  }, corePluginList)

  return result
}

export default function resolveConfig(configs) {
  let [plugins, pluginConfigs] = resolvePlugins(configs)

  let allConfigs = [
    ...pluginConfigs,
    {
      prefix: '',
      important: false,
      separator: ':',
    },
  ]

  return normalizeConfig(
    defaults(
      {
        theme: resolveFunctionKeys(
          mergeExtensions(mergeThemes(allConfigs.map((t) => t?.theme ?? {})))
        ),
        corePlugins: resolveCorePlugins(allConfigs.map((c) => c.corePlugins)),
        plugins,
      },
      ...allConfigs
    )
  )
}
