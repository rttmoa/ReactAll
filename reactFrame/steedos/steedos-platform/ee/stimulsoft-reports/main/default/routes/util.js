const objectql = require('@steedos/objectql');
const broker = objectql.getSteedosSchema().broker;
const _ = require('lodash');

/**
 * 
 * @param {*} query 查询设计器中的配置，query.options.parameters 获取参数默认值。
 * @param {*} report 报表记录。
 * @param {*} params 按钮中的执行脚本中带的参数。
 * @param {*} userId 
 * @param {*} spaceId 
 * @returns 
 */
const getQueryParameters = async (query, report, params, userSession) => {
    const parameters = {};
    const queryDefParameters = {};
    if (query && query.options && query.options.parameters) {
        _.each(query.options.parameters, (item) => {
            queryDefParameters[item.name] = _.isArray(item.value) ? item.value : item.value
        })
    }
    if (report.parameters && _.isArray(report.parameters)) {
        for (const parameter of report.parameters) {
            let value = params && _.has(params, parameter.name) ? params[parameter.name] : queryDefParameters[parameter.name];
            try {
                value = await objectql.computeSimpleFormula(parameter.value, params, userSession);
            } catch (error) {

            }
            parameters[parameter.name] = value;
        }
    }
    return parameters;
}

const getDatabase = async (reportId, userSession, params) => {
    const result = {
        serviceName: '',
        sampleConnectionString: '',
        data: {},
        types: {},
    }
    try {
        const record = await objectql.getObject('stimulsoft_reports').findOne(reportId);
        result.serviceName = record.label;
        result.sampleConnectionString = record.name;

        let queries = record.query;
        if (_.isString(record.query)) {
            queries = [record.query]
        }
        for (const queryId of queries) {
            const queryRecord = await broker.call('~packages-@steedos/service-charts.getQuery', { recordId: queryId });
            const parameters = await getQueryParameters(queryRecord, record, params || {}, userSession);
            const data = await broker.call('~packages-@steedos/service-charts.queries', { recordId: queryId, parameters, max_age: 10 });
            if (data && data.query_result && data.query_result.query) {
                const tableId = data.query_result.id || data.query_result._id;
                const dataSourceType = data.query_result.data_source_type || 'mongodb';
                try {
                    const tableColumns = {};
                    let objectFields = null;
                    if (dataSourceType === 'mongodb') {
                        const query = JSON.parse(data.query_result.query);
                        if (query.collection) {
                            tableColumns['_id'] = 'string';
                            const objectConfig = await objectql.getObject(query.collection).toConfig();
                            if (objectConfig) {
                                objectFields = objectConfig.fields;
                            }
                        }
                    }
                    _.each(data.query_result.data.columns, (item) => {
                        const name = item.name;
                        const type = item.type || 'string'
                        //TODO 如果字段在对象中，则自动根据对象中的字段类型，设置type
                        if (name.indexOf('.') < 0) {
                            const field = objectFields ? objectFields[name] : null;
                            if (field && field.multiple) {
                                // tableColumns[name] = 'Stimulsoft.System.StimulsoftStringList'
                                tableColumns[name] = 'string'
                            } else {
                                const ftype = field ? field.type : 'text';
                                switch (ftype) {
                                    case 'toggle':
                                    case 'boolean':
                                        tableColumns[name] = 'Stimulsoft.System.NullableBoolean'
                                        break;
                                    case 'number':
                                    case 'currency':
                                    case 'percent':
                                        tableColumns[name] = 'Stimulsoft.System.NullableLong'
                                        break;
                                    default:
                                        tableColumns[name] = type
                                        break;
                                }
                            }
                        }
                    })
                    result.types[tableId] = tableColumns;
                    _.each(data.query_result.data.rows, (row) => {
                        _.each(row, (v, k) => {
                            if (_.isArray(v)) {
                                row[k] = JSON.stringify(v)
                            }
                        })
                    })
                    result.data[tableId] = data.query_result.data.rows;
                } catch (error) {
                    console.log(error);
                }
            }
        }

    } catch (Exception) {
        console.log(Exception);

    }
    return result
}

exports.getDatabase = getDatabase;


exports.getReportWithData = async (reportId, userSession, parameters) => {
    const Stimulsoft = require('stimulsoft-reports-js');

    const record = await objectql.getObject('stimulsoft_reports').findOne(reportId);

    Stimulsoft.Base.StiFontCollection.addOpentypeFontFile("Roboto-Black.ttf");

    var report = new Stimulsoft.Report.StiReport();

    report.load(record.report_mrt);

    report.dictionary.databases.clear();

    const database = await getDatabase(reportId, userSession, parameters);

    var dataSet = new Stimulsoft.System.Data.DataSet(database.serviceName);

    dataSet.readJson(database.data);

    report.dictionary.dataSources.clear();

    _.each(database.types, (tableInfo, tableName) => {
        let dataSource = new Stimulsoft.Report.Dictionary.StiDataTableSource(
            database.serviceName,
            tableName
        );
        dataSource.sqlCommand = tableName;

        _.each(tableInfo, (columnType, columnName) => {
            dataSource.columns.add(
                new Stimulsoft.Report.Dictionary.StiDataColumn(
                    columnName,
                    columnName,
                    columnName,
                    columnType
                )
            );
        });
        report.dictionary.dataSources.add(dataSource);
    });

    report.regData(database.serviceName, database.serviceName, dataSet);

    return {
        report,
        data: database.data
    };
}