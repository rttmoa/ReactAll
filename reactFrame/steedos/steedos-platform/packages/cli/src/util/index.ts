export function getBaseDirectory(){
    //return require('app-root-path').path
    let cwd = process.cwd();
    if (cwd.indexOf('.meteor') > -1) {
        return cwd.split('.meteor')[0];
    }
    return cwd;
}

export function isObject(value) {
    return value != null && (typeof value === 'object' || typeof value === 'function');
}

export function isFunction(value) {
    return typeof value === 'function';
}

export function isPlainObject(value) {
    const isObjectObject = (o) => isObject(o) && Object.prototype.toString.call(o) === '[object Object]';
    if (!isObjectObject(value))
        return false;
    const ctor = value.constructor;
    if (!isFunction(ctor))
        return false;
    if (!isObjectObject(ctor.prototype))
        return false;
    if (!ctor.prototype.hasOwnProperty('isPrototypeOf'))
        return false;
    return true;
}

export function isJsonMap(value){
    return isPlainObject(value);
}