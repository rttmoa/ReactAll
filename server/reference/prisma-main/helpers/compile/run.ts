import execa from 'execa'

// wrapper around execa to run our build cmds

export function run(command: string) {
  return execa.command(command, {
    preferLocal: true,
    shell: true,
    stdio: 'inherit',
  })
}
