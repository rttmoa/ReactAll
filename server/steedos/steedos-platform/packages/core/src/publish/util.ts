const Fiber = require('fibers')
declare var DDP: any;
declare var DDPCommon: any;

export class Util{
    getCollectionMethod(name, target){
        return name.split('.').reduce(function(o, x) {
            return o[x];
        }, target);
    }

    async dbExec(userId, tableName, command, ...args: any[]){
        let collection = Creator.getCollection(tableName);
        let self = this;
        return await new Promise((resolve, reject) => {
            Fiber(function () {
                try {
                    let invocation = new DDPCommon.MethodInvocation({
                        isSimulation: true,
                        userId: userId,
                        connection: null,
                        randomSeed: DDPCommon.makeRpcSeed()
                    })
                    let result = DDP._CurrentInvocation.withValue(invocation, function () {
                        return self.getCollectionMethod(command, collection).apply(collection, args)
                    })
                    resolve(result);
                } catch (error) {
                    reject(error)
                }
            }).run()
        });
    }
}