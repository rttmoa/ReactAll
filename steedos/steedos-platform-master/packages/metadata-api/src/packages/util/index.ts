export * from './upload';
export * from './install';