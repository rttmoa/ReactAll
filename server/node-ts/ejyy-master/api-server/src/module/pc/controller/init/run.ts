/**
 * +----------------------------------------------------------------------
 * | 「e家宜业」 —— 助力物业服务升级，用心服务万千业主
 * +----------------------------------------------------------------------
 * | Copyright (c) 2020-2022 https://www.chowa.cn All rights reserved.
 * +----------------------------------------------------------------------
 * | Licensed 未经许可不能去掉「e家宜业」和「卓瓦科技」相关版权
 * +----------------------------------------------------------------------
 * | Author: contact@chowa.cn
 * +----------------------------------------------------------------------
 */

import { Action } from '~/types/action';
import { SUCCESS, SYSTEMT_ALREADY_INIT } from '~/constant/code';
import config from '~/config';
import { TRUE, FALSE } from '~/constant/status';
import utils from '~/utils';

interface RequestBody {
    name: string;
    banner: string;
    province: string;
    city: string;
    district: string;
    service_phone: string;
    access_nfc: typeof TRUE | typeof FALSE;
    access_remote: typeof TRUE | typeof FALSE;
    access_qrcode: typeof TRUE | typeof FALSE;
    carport_max_car: number;
    fitment_pledge: typeof TRUE | typeof FALSE;
    // profile
    account: string;
    password: string;
    real_name: string;
    avatar_url: string;
    phone: string;
    idcard: string;
}

const PcInitRunAction = <Action>{
    router: {
        path: '/init/run',
        method: 'post',
        authRequired: false
    },
    validator: {
        body: [
            {
                name: 'name',
                required: true,
                max: 12
            },
            {
                name: 'banner',
                required: true,
                max: 128,
                regex: /^\/community_banner\/[a-z0-9]{32}\.(jpg|jpeg|png)$/
            },
            {
                name: 'province',
                required: true,
                max: 12
            },
            {
                name: 'city',
                required: true,
                max: 12
            },
            {
                name: 'district',
                required: true,
                max: 12
            },
            {
                name: 'service_phone',
                required: true,
                length: 11
            },
            {
                name: 'access_nfc',
                required: true,
                validator: val => [TRUE, FALSE].includes(val)
            },
            {
                name: 'access_remote',
                required: true,
                validator: val => [TRUE, FALSE].includes(val)
            },
            {
                name: 'access_qrcode',
                required: true,
                validator: val => [TRUE, FALSE].includes(val)
            },
            {
                name: 'carport_max_car',
                required: true,
                regex: /^\d+$/
            },
            {
                name: 'fitment_pledge',
                required: true,
                validator: val => [TRUE, FALSE].includes(val)
            },
            {
                name: 'account',
                required: true,
                min: 4,
                max: 32
            },
            {
                name: 'password',
                required: true,
                max: 32
            },
            {
                name: 'real_name',
                required: true,
                max: 8
            },
            {
                name: 'idcard',
                required: true,
                validator: val => utils.idcard.verify(val)
            },
            {
                name: 'avatar_url',
                required: true,
                max: 128,
                validator: val => /^\/avatar\/[a-z0-9]{32}|default\.(jpg|jpeg|png)$/.test(val)
            },
            {
                name: 'phone',
                required: true,
                regex: /^1\d{10}$/
            }
        ]
    },
    response: async ctx => {
        const {
            name,
            banner,
            province,
            city,
            district,
            service_phone,
            access_nfc,
            access_remote,
            access_qrcode,
            carport_max_car,
            fitment_pledge,
            account,
            password,
            real_name,
            idcard,
            avatar_url,
            phone
        } = <RequestBody>ctx.request.body;

        const total = utils.sql.countReader(
            await ctx.model
                .from('ejyy_property_company_user')
                .where('admin', TRUE)
                .count()
        );

        if (total > 0) {
            return (ctx.body = {
                code: SYSTEMT_ALREADY_INIT,
                message: '系统已经初始化'
            });
        }

        const created_at = Date.now();

        const [user_id] = await ctx.model.from('ejyy_property_company_user').insert({
            account,
            password: utils.crypto.md5(password),
            real_name,
            idcard,
            gender: utils.idcard.gender(idcard),
            avatar_url,
            phone,
            admin: TRUE,
            join_company_at: created_at,
            created_at,
            leave_office: FALSE
        });

        const [community_id] = await ctx.model.from('ejyy_community_info').insert({
            name,
            banner,
            province,
            city,
            district,
            phone: service_phone,
            created_at,
            created_by: user_id
        });

        await ctx.model.from('ejyy_community_setting').insert({
            access_nfc,
            access_remote,
            access_qrcode,
            carport_max_car,
            fitment_pledge,
            community_id
        });

        await ctx.model.from('ejyy_property_company_user_access_community').insert({
            community_id,
            property_company_user_id: user_id
        });

        await ctx.model.from('ejyy_property_company_auth').insert({
            property_company_user_id: user_id,
            token: null
        });

        config.inited = true;

        ctx.body = {
            code: SUCCESS,
            message: '系统初始化成功'
        };
    }
};

export default PcInitRunAction;
