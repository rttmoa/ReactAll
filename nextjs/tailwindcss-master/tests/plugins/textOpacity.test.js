import { quickPluginTest } from '../util/run'

quickPluginTest('textOpacity', {
  safelist: [
    // Arbitrary values
    'text-opacity-[12%]',
    'text-opacity-[var(--my-value)]',
  ],
  theme: {
    opacity: {
      0: '0',
      50: '.5',
    },
  },
}).toMatchSnapshot()
