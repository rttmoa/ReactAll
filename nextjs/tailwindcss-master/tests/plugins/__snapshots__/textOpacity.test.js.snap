// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the 'textOpacity' plugin 1`] = `
"
.text-opacity-0 {
  --tw-text-opacity: 0;
}

.text-opacity-50 {
  --tw-text-opacity: .5;
}

.text-opacity-\\[12\\%\\] {
  --tw-text-opacity: 12%;
}

.text-opacity-\\[var\\(--my-value\\)\\] {
  --tw-text-opacity: var(--my-value);
}
"
`;
